### PR TITLE
[Estuary] Hide episode thumb if unwatched

### PR DESCRIPTION
**Pull request by DaVukovic**
## Description

This change makes it possible to hide the episode thumb depending on the plot. If the plot is `* Hidden to prevent spoilers *` the `OverlaySpoiler.png` image will be shown as an overlay of the actual thumb.

That image is an 85% black image combined with the "DefaultActor.png". That was the best match I could find style-wise and the mask fits best for that part, I would say. I'm open to other suggestions, but I'm not a designer ;)

## Motivation and Context

That was kind of "per request" from @MartijnKaijser and I also like the idea. As we have the option to seperate which kind of plot we would like to show from unwatched items, I also thought that it would be a good idea to hide the thumb as well. 

## How Has This Been Tested?

I've tested any possible view for the episode listing. Also the DialogVideoInfo.xml ("i" keypress on the keyboard) hides the thumb if the episode is unwatched and at the home screen, recently added episode thumbs are also hidden now. 

## Screenshots (if appropriate):

As screenshots will say more than 1000 words...

watched1
watched2
watched3
watched4

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by hudokkow** [comment]
Looks great!

However, I think the icon/mask could be improved. Something along the lines of


anonymous svg


**credit:** https://commons.wikimedia.org/wiki/File:Anonymous.svg#/media/File:Anonymous.svg


:smile:

**Comment by ronie** [comment]
what about doing it in core instead?
see: https://github.com/notspiff/kodi-cmake/commit/9ec02c114fa19e76e25a99699f298849466ffe6b 

since the plot is hidden by core, it'd make sense to do the same for the thumb i think...


in case you prefer doing it skin side, please use includes instead of visible conditions.

**Review by Hitcher — APPROVED** [review]
Looks good to me, thanks.

EDIT: Now that ronie mentioned it - a core solution would be ideal and would only need a new default icon added to skins by the look of that code.

EDIT 2: lol at the date.

**Comment by DaVukovic** [comment]
@ronie I would have done that in core, but I'm not a c++ guy. So doing that is pretty hard for me, tbh ;)

I'm 100% fine if it would be better at core and if we would drop this PR instead. I'll look into the link you send me and see what I can do but I can't promise anything.

If we decide to do it skin side, I'll also take a look how to use includes instead of conditions.

**Comment by MartijnKaijser** [comment]
I'd rather have it core cause then its global and not skin dependable

**Comment by HitcherUK** [comment]
> I would have done that in core, but I'm not a c++ guy. So doing that is pretty hard for me, tbh ;)
> 
> I'm 100% fine if it would be better at core and if we would drop this PR instead. I'll look into the link you send me and see what I can do but I can't promise anything.

Can spiff help?

File: addons/skin.estuary/xml/DialogVideoInfo.xml [hunk 1]
```diff
@@ -53,6 +53,17 @@
...
+					<control type="image">
+						<left>14</left>
+						<top>4</top>
+						<width>506</width>
+						<height>801</height>
+						<aspectratio>keep</aspectratio>
+						<texture>OverlaySpoiler.png</texture>
+						<bordertexture border="21">overlays/shadow.png</bordertexture>
+						<bordersize>20</bordersize>
+						<visible>String.IsEqual(ListItem.Plot,$LOCALIZE[20370])</visible>
```
Comment (ronie) [inline addons/skin.estuary/xml/DialogVideoInfo.xml]: it would be better use this condition: System.GetBool(videolibrary.showunwatchedplots)

**Comment by HitcherUK** [comment]
If this can be added to core alongside 'Show plot for unwatched items' can I suggest we rename the whole thing 'Show spoilers for unwatched items' and make each one (movie plot, episode plot, and episode thumb) separate options. For me at least episode thumbs are the biggest spoiler threat.

File: addons/skin.estuary/xml/DialogVideoInfo.xml (hunk 1 above)
Comment (DaVukovic) [reply addons/skin.estuary/xml/DialogVideoInfo.xml]: This setting is not a bool anymore: 

https://github.com/xbmc/xbmc/blob/a45d319f0ae8b3afde07d1b6b7ed9a77e4e2fd33/system/settings/settings.xml#L971

**Comment by DaVukovic** [comment]
@HitcherUK The idea is not that bad and I will make my thoughts how to achieve that. 

We could either add `episode thumbs` to the setting from 

https://github.com/xbmc/xbmc/pull/14489

Or we create a subsetting which depends on the "Show plot/spoiler" setting which could itself be a bool again then. 

OTOH I guess we don't want to fill Estuary with more and more settings. So adding to an existing setting might probably a better idea.

File: addons/skin.estuary/xml/DialogVideoInfo.xml (hunk 1 above)
Comment (ronie) [reply addons/skin.estuary/xml/DialogVideoInfo.xml]: ah, wasn't aware.. ignore :-)

**Comment by jjd-uk** [comment]
Note there were some issues found during the last attempt to do this in core, see https://github.com/xbmc/xbmc/pull/7462

Of course those issues may not longer be found in the current code.

**Comment by DaVukovic** [comment]
As I guess the focus is way more on #14971 it's fine to close this one then. I'll keep the branch until the other one is merged

**Pull request closed without merging**